backend/api/Services/InspectionFindingsService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Api.Database.Context;
+using Microsoft.EntityFrameworkCore;
+
+
+namespace Api.Services
+{
+    public class InspectionFindingService(
+        ILogger<InspectionFindingService> logger,
+        IServiceScopeFactory serviceScopeFactory,
+        IHostApplicationLifetime hostApplicationLifetime,
+        IConfiguration configuration) : IHostedService
+    {
+        private readonly ILogger<InspectionFindingService> _logger = logger;
+        private readonly IServiceScopeFactory _serviceScopeFactory = serviceScopeFactory;
+        private readonly IHostApplicationLifetime _hostApplicationLifetime = hostApplicationLifetime;
+        private readonly TimeSpan _timeSpan = configuration.GetValue<TimeSpan>("InspectionFindingEventHandler:TimeSpan");
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        public async Task GenerateInspectionReports()
+        {
+            var lastReportingTime = DateTime.UtcNow - _timeSpan;
+
+            try
+            {
+                using var scope = _serviceScopeFactory.CreateScope();
+                var context = scope.ServiceProvider.GetRequiredService<FlotillaDbContext>();
+
+                var inspectionFindings = await context.InspectionFindings
+                                            .Where(f => f.InspectionDate > lastReportingTime)
+                                            .ToListAsync();
+
+                _logger.LogInformation("Found {count} inspection findings in the last {interval}.", inspectionFindings.Count, _timeSpan);
+
+                foreach (var finding in inspectionFindings)
+                {
+                    _logger.LogInformation("- Id: {id}", finding.Id);
+                    _logger.LogInformation("  InspectionDate: {date}", finding.InspectionDate);
+                    _logger.LogInformation("  IsarStepId: {id}", finding.IsarStepId);
+                    _logger.LogInformation("  Findings: {findings}", finding.Findings);
+                }
+
+                // check if the application is stopping, and exit the loop if necessary
+                if (_hostApplicationLifetime.ApplicationStopping.IsCancellationRequested)
+                {
+                    _logger.LogInformation("Application is stopping. Terminating InspectionFindingService.");
+
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error in executing InspectionFindingService; {Message}", ex.Message);
+            }
+        }
```

Comment:
I can simply remove the log information of inspection findings, but I will still consider keeping the try and catch block since that is good practice, no ?

Reply:
Done.